workflow-ui/core-common/src/main/java/com/squareup/workflow1/ui/container/EnvironmentScreen.kt

```diff
@@ -47,13 +49,14 @@ public fun Screen.withRegistry(viewRegistry: ViewRegistry): EnvironmentScreen<*>
  * to preserve the [ViewRegistry] entries of both.
  */
 @WorkflowUiExperimentalApi
-public fun Screen.withEnvironment(
+public fun <V : Screen> V.withEnvironment(
   environment: ViewEnvironment = ViewEnvironment.EMPTY
-): EnvironmentScreen<*> {
+): EnvironmentScreen<V> {
```

Comment:
Isn't this going to erroneously claim to return `EnvironmentScreen<EnvironmentScreen<*>>` when the receiver is already one?

Reply:
Yep... 🤦 

I've got to come up with something else.